Content.Server/Devour/DevouredSystem.cs

```diff
@@ -0,0 +1,136 @@
+            entity.Comp.OriginalAllowedMobStates = passiveDamage.AllowedStates;
+
+        //Stores the stomach damage value if it exists.
+        if (TryComp<DevourerComponent>(args.Devourer, out var devourer))
+        {
+            if (devourer.StomachDamage != null)
+                entity.Comp.StomachDamage = devourer.StomachDamage;
+
+            //Sets the MobStates in which the devoured entity is damaged.
+            passiveDamage.AllowedStates = devourer.DigestibleStates;
+        }
+
+        //Sets the damage multiplier based on MobState.
+        if (TryComp<MobStateComponent>(entity, out var mobState))
+        {
+            SetStomachDamage(mobState.CurrentState, entity.Comp, passiveDamage);
+        }
+
+        //Sets the max damage to the damage needed for the mob to be considered dead +
+        //the damage cap specified in the component. This can be used to make reviving take more work if
+        //the target has been devoured for a long time.
+        if (TryComp<MobThresholdsComponent>(entity, out var mobThresholds))
+        {
+            for (var mobStates = 0; mobStates < mobThresholds.Thresholds.Count; mobStates++)
+            {
+                var mobStateValue = mobThresholds.Thresholds.ElementAt(mobStates);
+                if (mobStateValue.Value == MobState.Dead)
+                {
+                    passiveDamage.DamageCap = mobStateValue.Key + entity.Comp.DamageCap;
+                    break;
+                }
+            }
+        }
+
+        //Pacifies entities in the stomach and stores if the target was already pacified.
+        //This is to make sure the pacified component won't be removed if the entity
+        //already had it before being devoured.
+        if (EnsureComp<PacifiedComponent>(entity, out var pacified))
+            entity.Comp.OriginallyPacified = true;
+    }
+
+    /// <summary>
+    ///     Removes the effects that were added when the entity was devoured.
+    /// </summary>
+    private void OnRemove(EntityUid uid, DevouredComponent component, ComponentRemove args)
+    {
+        //Remove the component if the target originally didn't have it.
+        if (component.OriginalAllowedMobStates == null)
+        {
+            RemComp<PassiveDamageComponent>(uid);
+        }
+        else
+        {
+            //Return the passive damage component back to what it was before the entity got devoured.
+            if (TryComp<PassiveDamageComponent>(uid, out var passiveDamage) && component.StomachDamage != null)
+            {
+                passiveDamage.Damage -= component.StomachDamage * component.CurrentModifier;
+                passiveDamage.AllowedStates = component.OriginalAllowedMobStates;
+                passiveDamage.Damage.TrimZeros();
+            }
+        }
+
+        //Remove the pacified component if the target originally didn't have it.
+        if (!component.OriginallyPacified)
+            RemComp<PacifiedComponent>(uid);
```

Comment:
Why is this needed?

Reply:
It's so people that had the pacified component before they got devoured(thieves and people with the pacifist trait) will still have the component. But I'll change it to actionblocker like you suggested.